lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -20,13 +20,17 @@ function wp_theme_has_theme_json() {
 		$theme_has_support = wp_cache_get( $cache_key, $cache_group );
 
 		/**
-		 * $theme_has_support is stored as a int in the cache.
+		 * $theme_has_support is stored as an int in the cache.
 		 *
 		 * The reason not to store it as a boolean is to avoid working
 		 * with the $found parameter which apparently had some issues in some implementations
 		 * https://developer.wordpress.org/reference/functions/wp_cache_get/
 		 */
-		if ( 0 === $theme_has_support || 1 === $theme_has_support ) {
+		if (
+			// Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
+			( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
+			( 0 === $theme_has_support || 1 === $theme_has_support )
```

Comment:
```suggestion
			( is_int( $theme_has_support ) )
```

Reply:
I'd rather be explicit and check for 0/1, the only valid values.

Reply:
This is technically not part of this PR, since this check was added on a previous PR (and it doesn't change with this PR). I'm happy to commit this though, but want to ask @oandregal before if there is any reason why he added a strict check against `0`/`1` instead of a more generic `is_int` check.